src/js/plugins/plugin.print.js

```diff
@@ -0,0 +1,82 @@
+/*global BookReader */
+
+/**
+ * Adds some print features to bookreader
+ *
+ * To bind it to a button, use code like this:
+ *     jIcons.filter('.print').click(function(e) {
+ *         self.printPage();
+ *       return false;
+ *   });
+ */
+
+/**
+ * Plugin which used to print book images.
+ */
+jQuery.extend(BookReader.defaultOptions, {
+  imageFormat: 'jp2',
+  subPrefix: '',
+  server: '',
+  zip: '',
+});
+
+/**
+ * @augments BookReader.setup
+ */
+BookReader.prototype.setup = (function (super_) {
+  return function (options) {
+    super_.call(this, options);
+
+    this.imageFormat = options.imageFormat;
+    this.subPrefix = options.subPrefix;
+    this.server = options.server;
+    this.zip = options.zip;
+  };
+})(BookReader.prototype.setup);
+
+
+BookReader.prototype.printPage = function() {
+  window.open(
+    this.getPrintURI(),
+    'printpage',
+    'width=400, height=500, resizable=yes, scrollbars=no, toolbar=no, location=no'
+  );
+};
+
+/** 
+ * Get print URI from current indices and mode
+ */
+BookReader.prototype.getPrintURI = function() {
+  var indexToPrint;
+  if (this.constMode2up === this.mode) {
+    indexToPrint = this.twoPage.currentIndexL;
+  } else {
+    indexToPrint = this.firstIndex; // $$$ the index in the middle of the viewport would make more sense
+  }
+
+  var options = 'id=' + this.subPrefix + '&server=' + this.server + '&zip=' + this.zip
+    + '&format=' + this.imageFormat + '&file=' + this.getPageFile(indexToPrint)
+    + '&width=' + this._getPageWidth(indexToPrint) + '&height=' + this._getPageHeight(indexToPrint);
```

Comment:
You could potentially change this to use ES6 string interpolation syntax, but if you do, please write some tests beforehand since it's a somewhat complex string concatenation.

Reply:
I am leaving this for now as seems like we are not using this plugin right now. Once we know where we are using this, will focus on this.